pennylane/gradients/jvp.py

```diff
@@ -295,11 +295,16 @@ def jvp(tape, tangent, gradient_fn, gradient_kwargs=None):
     if len(tape.trainable_params) == 0:
         # The tape has no trainable parameters; the JVP
         # is simply none.
-        def zero_vjp(_):
-            res = tuple(np.zeros(mp.shape(None, tape.shots)) for mp in tape.measurements)
+        def zero_jvp_for_single_shots(s):
+            res = tuple(np.zeros(mp.shape(shots=s)) for mp in tape.measurements)
```

Comment:
Do we also need to add in the numeric type?
```suggestion
            res = tuple(np.zeros(mp.shape(shots=s), dtype=mp.numeric_type) for mp in tape.measurements)
```

Reply:
Ah good point!